### PR TITLE
update flameshot script to be universal

### PR DESCRIPTION
**Pull request by NotAShelf**
- wayland support
- experimental local screenshots if domain & key are left empty

## Checklist

- [x] I have read the [Contributing Guidelines]
- [x] I acknowledge that any submitted code will be licensed under the [ISC License]
- [x] I confirm that submitted code is my own work (partially)
- [x] I have tested the code, and confirm that it works

## Enviroment
- Operating System: NixOS 6.5.5-xanmod1
- Tested On: only NixOS

## Description

The script now supports Wayland through the help of Wayland native tools (such as grim and slurp) instead of attempting to get Flameshot to work on [Sway and other wl-roots based compositors](https://github.com/flameshot-org/flameshot/blob/master/docs/Sway%20and%20wlroots%20support.md). 

Additionally, it fixes a possible bug that went unnoticed with the [`if [[ -f $FILE ]]` condition check](https://github.com/tycrek/ass/blob/master/flameshot_example.sh#L14=). 

Lastly, support for **local** screenshots are added and any screenshot taken with the script while `KEY` and `DOMAIN` values are null will be copied to the clipboard directly instead of throwing an error and exiting the script.

**tl;dr:**
- wayland support
- error turned into feature (local screenshots)
- bug fix
- logic and syntax improvements
- script is now more verbose

**Dependencies:** (Wayland)
- [grim](https://sr.ht/~emersion/grim/)
- [slurp](https://github.com/emersion/slurp)
- [wl-clipboard](https://github.com/bugaevc/wl-clipboard)

[Contributing Guidelines]: https://github.com/tycrek/ass/blob/master/CONTRIBUTING.md
[ISC License]: https://github.com/tycrek/ass/blob/master/LICENSE

### CODE REVIEW

**Comment by samip5** [comment]
I have been using this pretty much daily and works great. Thank you. :)

**Comment by tycrek** [comment]
I'll aim to merge this with ass 0.15.0 and the universal script that exists on that branch.

**Comment by NotAShelf** [comment]
> I'll aim to merge this with ass 0.15.0 and the universal script that exists on that branch.

I'll do some tweaking to the script, please do give me a headsup before merging

**Comment by Gauvino** [comment]
@tycrek https://github.com/tycrek/ass/pull/135#issuecomment-1762988176

**Comment by NotAShelf** [comment]
should be ready now

**Review by tycrek — APPROVED** [review]
Looks awesome; I'll merge in for now and combine it with the `flameshot-v2.sh` script at some point.
